include/albatross/src/stats/ks_test.hpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef ALBATROSS_STATS_KS_TEST_HPP_
+#define ALBATROSS_STATS_KS_TEST_HPP_
+
+namespace albatross {
+
+/*
+ * Computes the two-sided Kolmogorov-Smirnov test to determine
+ * if the samples came from a uniform distribution.
+ */
+inline double uniform_ks_test(const std::vector<double> &samples) {
+  double largest_difference_between_sample_and_expected_cdf = 0.;
+  std::vector<double> sorted(samples);
+  std::sort(sorted.begin(), sorted.end());
+  double n = static_cast<double>(sorted.size());
+  for (std::size_t i = 0; i < sorted.size(); ++i) {
+
+    double difference = std::max((i + 1) / n - sorted[i], sorted[i] - i / n);
```

Comment:
I would've expected something symmetric here like `fabs(  sorted[i] -  i/(n-1)  )`, but doing n=1 and n=2 in my head, this works out better.

Reply:
Yeah I originally an abs in there ... then switched after looking at the python implementation:

https://github.com/scipy/scipy/blob/bf3c6471aba06cf6a96843d62f5198ec90e15cf1/scipy/stats/stats.py#L4868

which does what I'm doing here (for the "two_sided") mode.  I'm not totally convinced it's better, but sorta assume enough people use that implementation that it must be somewhat weather proven.

Reply:
I worded my first comment poorly. The python/PR implementation is definitely better for small N examples, and equivalent for large N.